src/execution_plan/optimizations/utilize_indices.c

```diff
@@ -181,241 +97,75 @@ bool _simple_predicates(FT_FilterNode *filter) {
-	char     *field  =  NULL;         // field being filtered
-	SIValue  origin  =  SI_NullVal(); // center of circle
-	SIValue  radius  =  SI_NullVal(); // circle radius
-
-	_extractOriginAndRadius(filter, &origin, &radius, &field);
-
-	return RediSearch_CreateGeoNode(idx, field, Point_lat(origin),
-									Point_lon(origin), SI_GET_NUMERIC(radius), RS_GEO_DISTANCE_M);
-}
-
-// Creates a RediSearch query node out of given IN filter.
-RSQNode *_filterTreeToInQueryNode(FT_FilterNode *filter, RSIndex *sp) {
-	ASSERT(_isInFilter(filter));
-
-	// n.v IN [1,2,3]
-	// a single union node should hold a number of token/numeric nodes
-	// one for each element in the array.
-
-	// extract both field name and list from expression
-	AR_ExpNode *inOp = filter->exp.exp;
-
-	char *field;
-	bool attribute = AR_EXP_IsAttribute(inOp->op.children[0], &field);
-	ASSERT(attribute == true);
-
-	SIValue list = inOp->op.children[1]->operand.constant;
-	uint list_len = SIArray_Length(list);
-
-	if(list_len == 0) {
-		// Special case: "WHERE a.v in []"
-		return RediSearch_CreateEmptyNode(sp);
-	}
-
-	RSQNode *node = NULL;
-	RSQNode *parent = NULL;
-	RSQNode *U = RediSearch_CreateUnionNode(sp);
-
-	for(uint i = 0; i < list_len; i ++) {
-		double d;
-		SIValue v = SIArray_Get(list, i);
-		switch(SI_TYPE(v)) {
-			case T_STRING:
-				parent = RediSearch_CreateTagNode(sp, field);
-				node = RediSearch_CreateTokenNode(sp, field, v.stringval);
-				RediSearch_QueryNodeAddChild(parent, node);
-				node = parent;
-				break;
-			case T_DOUBLE:
-			case T_INT64:
-			case T_BOOL:
-				d = SI_GET_NUMERIC(v);
-				node = RediSearch_CreateNumericNode(sp, field, d, d, true, true);
-				break;
-			default:
-				ASSERT(false && "unexpected conditional operation");
-				break;
 		}
-		RediSearch_QueryNodeAddChild(U, node);
-	}
-
-	return U;
-}
 
-// Creates a RediSearch query node out of given filter tree.
-RSQNode *_filterTreeToQueryNode(FT_FilterNode *filter, RSIndex *sp) {
-	RSQNode *node = NULL;
-	RSQNode *parent = NULL;
-
-	switch(filter->t) {
-		case FT_N_COND: {
-			RSQNode *left = NULL;
-			RSQNode *right = NULL;
-			switch(filter->cond.op) {
-				case OP_OR:
-					node = RediSearch_CreateUnionNode(sp);
-					left = _filterTreeToQueryNode(filter->cond.left, sp);
-					right = _filterTreeToQueryNode(filter->cond.right, sp);
-					RediSearch_QueryNodeAddChild(node, left);
-					RediSearch_QueryNodeAddChild(node, right);
-					break;
-				case OP_AND:
-					node = RediSearch_CreateIntersectNode(sp, false);
-					left = _filterTreeToQueryNode(filter->cond.left, sp);
-					right = _filterTreeToQueryNode(filter->cond.right, sp);
-					RediSearch_QueryNodeAddChild(node, left);
-					RediSearch_QueryNodeAddChild(node, right);
-					break;
-				default:
-					ASSERT(false && "unexpected conditional operation");
-					break;
-			}
+		if(AR_EXP_IsAttribute(lhs_exp, NULL)) exp = rhs_exp; // n.v = exp
+		if(AR_EXP_IsAttribute(rhs_exp, NULL)) exp = lhs_exp; // exp = n.v
```

Comment:
I feel like `else if` would be more natural here.

Reply:
right